### DISCUSSION TITLE
Screenshot Action VS script

### DISCUSSION THREAD

**Discussion by riekusr** [drudge/n8n-nodes-puppeteer#18] [General]
Hi,

Firstly, THANK YOU! 

I want to make a screenshot of the page after it is rendered which I am doing with the "Run Custom script" node. How can I do either one of the following: 

option 1 - Use the 'screenshot' node, but then don't use the URL but the current state of the browser 'page'? 

option 2 - Use the   await page.screenshot({path: 'screenshot.png'}); in me "Run Custom Script" node but then also have the image retuned in the 'output' viewable of N8N editor? Preferably with other output, like: 

{
screenshot: screenshot, 
data: data
}

and again... THANK YOU!!!!

**Comment by drudge** [comment]
Hey @riekusr thanks for your interest in my n8n node. I've released an update that allows you to add binary property data so you can return it in your custom script.  See the v1.1.0 release notes for more details.

```js
await $page.goto('https://www.google.com')

const imageData = await $page.screenshot({ type: 'png', encoding: 'base64' })

return [
  {
    binary: {
      screenshot: {
        data: imageData,
        mimeType: 'image/png',
        fileName: 'screenshot.png',
      },
    },
  }
]
```
